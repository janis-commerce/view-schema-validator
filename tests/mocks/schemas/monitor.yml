service: picking
name: views-demo-monitor
root: Monitor

autoRefresh: true

source:
  service: playground
  namespace: views-demo
  method: list
  resolve: false

endpointParameters:
  - name: status
    target: filter
    value:
      static: active

schemaSource:
  type: dynamic
  endpoint:
    service: playground
    namespace: views-demo
    method: monitor
    resolve: false

themes:
  themeOne:
    new: black
    closed: green
    _default: grey
  themeTwo:
    new: grey
    closed: fizzgreen
    warning:
      somePropOne: someValue
      somePropTwo: someValue

statusBar:
  field: statusFieldName
  hide: true
  useTheme: true

sortableFields:
  - name: test
  - name: test1
    isDefaultSort: true
  - name: test2
    initialSortDirection: asc

filters:
  - name: filterInput
    component: Input
    componentAttributes:
      icon: iconName

  - name: localSelect
    component: Select
    componentAttributes:
      translateLabels: true
      options:
        scope: local
        values:
          - label: test
            value: 1

dependencies:
  - name: dependencyOne
    source:
      service: serviceName
      namespace: namespaceName
      method: methodName
      resolve: false
    endpointParameters:
      - name: status
        target: path
        value:
          dynamic: id
      - name: status
        target: filter
        value:
          static: active
    targetField: fieldNameOne
  - name: dependencyTwo
    source:
      service: serviceName
      namespace: namespaceName
      method: methodName
      resolve: false
    endpointParameters:
      - name: status
        target: path
        value:
          dynamic: id
      - name: status
        target: filter
        value:
          static: active
    targetField: fieldNameTwo
    dependencies:
      - name: dependencyThree
        source:
          service: serviceName
          namespace: namespaceName
          method: methodName
          resolve: false
        endpointParameters:
          - name: status
            target: path
            value:
              dynamic: id
          - name: status
            target: filter
            value:
              static: active
        targetField: fieldNameThree

fields:
  - name: someCardOne
    component: BaseCard
    label: some.label.key
    translateLabel: false
    actions:
      - name: testEndpoint
        callback: refresh
        endpoint:
          service: serviceName
          namespace: namespaceName
          method: methodName
          resolve: false
        endpointParameters:
          - name: id
            target: path
            value:
              dynamic: id

      - name: testLink
        path: /test/{id}
        target: _self
        conditions:
          showWhen:
            - - name: isEqualTo
                field: name
                referenceValue: test
        endpointParameters:
          - name: id
            target: path
            value:
              dynamic: id

      - name: testForm
        endpoint:
          service: serviceName
          namespace: namespaceName
          method: methodName
          resolve: false
        callback: reloadMonitorData
        schema:
          - name: fieldNameOne
            component: Input

          - name: fieldNameTwo
            component: Switch

    fieldsGroup:
      - name: info
        fields:
          - name: someField
            component: Text
            dependency: dependencyOne
          - name: otherField
            component: Chip
    conditions:
      matchWhen:
        - - name: isNotEmpty
            field:
              - test
              - tes2
          - name: isNotEqualTo
            field: name
            referenceValueType: static
            referenceValue: null

  - name: someCardTwo
    component: BaseCard
    fieldsGroup:
      - name: info
        fields:
          - name: someField
            component: Text
          - name: otherField
            component: Chip
    conditions:
      matchWhen:
        - - name: isEqualTo
            field: name
            referenceValue: test

