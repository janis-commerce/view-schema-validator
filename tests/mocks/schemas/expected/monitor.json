{
    "service": "picking",
    "name": "views-demo-monitor",
    "root": "Monitor",
    "autoRefresh": 60,
    "source": {
        "service": "playground",
        "namespace": "views-demo",
        "method": "list",
        "resolve": false
    },
    "endpointParameters": [
        {
            "name": "status",
            "target": "filter",
            "value": {
                "static": "active"
            }
        }
    ],
    "schemaSource": {
        "type": "dynamic",
        "endpoint": {
            "service": "playground",
            "namespace": "views-demo",
            "method": "monitor",
            "resolve": false
        }
    },
    "cardLink": {
        "path": "/route/{id}/edit",
        "endpointParameters": [
            {
                "name": "id",
                "target": "path",
                "value": {
                    "dynamic": "id"
                }
            }
        ]
    },
    "themes": {
        "themeOne": {
            "new": "black",
            "closed": "green",
            "_default": "grey"
        },
        "themeTwo": {
            "new": "grey",
            "closed": "fizzgreen",
            "warning": {
                "somePropOne": "someValue",
                "somePropTwo": "someValue"
            }
        }
    },
    "statusBar": {
        "field": "statusFieldName",
        "hide": true,
        "useTheme": true
    },
    "sortableFields": [
        {
            "name": "test",
            "isDefaultSort": false,
            "initialSortDirection": "desc"
        },
        {
            "name": "test1",
            "isDefaultSort": true,
            "initialSortDirection": "desc"
        },
        {
            "name": "test2",
            "isDefaultSort": false,
            "initialSortDirection": "asc"
        }
    ],
    "filters": [
        {
            "name": "filterInput",
            "component": "Input",
            "componentAttributes": {
                "icon": "iconName"
            }
        },
        {
            "name": "localSelect",
            "component": "Select",
            "componentAttributes": {
                "translateLabels": true,
                "options": {
                    "scope": "local",
                    "values": [
                        {
                            "label": "test",
                            "value": 1
                        }
                    ]
                }
            }
        }
    ],
    "dependencies": [
        {
            "name": "dependencyOne",
            "source": {
                "service": "serviceName",
                "namespace": "namespaceName",
                "method": "methodName",
                "resolve": false
            },
            "endpointParameters": [
                {
                    "name": "status",
                    "target": "path",
                    "value": {
                        "dynamic": "id"
                    }
                },
                {
                    "name": "status",
                    "target": "filter",
                    "value": {
                        "static": "active"
                    }
                }
            ],
            "targetField": "fieldNameOne"
        },
        {
            "name": "dependencyTwo",
            "source": {
                "service": "serviceName",
                "namespace": "namespaceName",
                "method": "methodName",
                "resolve": false
            },
            "endpointParameters": [
                {
                    "name": "status",
                    "target": "path",
                    "value": {
                        "dynamic": "id"
                    }
                },
                {
                    "name": "status",
                    "target": "filter",
                    "value": {
                        "static": "active"
                    }
                }
            ],
            "targetField": "fieldNameTwo",
            "dependencies": [
                {
                    "name": "dependencyThree",
                    "source": {
                        "service": "serviceName",
                        "namespace": "namespaceName",
                        "method": "methodName",
                        "resolve": false
                    },
                    "endpointParameters": [
                        {
                            "name": "status",
                            "target": "path",
                            "value": {
                                "dynamic": "id"
                            }
                        },
                        {
                            "name": "status",
                            "target": "filter",
                            "value": {
                                "static": "active"
                            }
                        }
                    ],
                    "targetField": "fieldNameThree"
                }
            ]
        }
    ],
    "fields": [
        {
            "name": "someCardOne",
            "component": "BaseCard",
            "label": "some.label.key",
            "translateLabel": false,
            "actions": [
                {
                    "name": "testEndpoint",
                    "type": "endpoint",
                    "callback": "refresh",
                    "componentAttributes": {
                        "icon": "box",
                        "endpoint": {
                            "service": "serviceName",
                            "namespace": "namespaceName",
                            "method": "methodName",
                            "resolve": false
                        },
                        "endpointParameters": [
                            {
                                "name": "id",
                                "target": "path",
                                "value": {
                                    "dynamic": "id"
                                }
                            }
                        ]
                    }
                },
                {
                    "name": "testLink",
                    "type": "link",
                    "conditions": {
                        "showWhen": [
                            [
                                {
                                    "name": "isEqualTo",
                                    "field": "orderData",
                                    "innerField": "status",
                                    "referenceValue": "active"
                                }
                            ]
                        ]
                    },
                    "componentAttributes": {
                        "path": "/test/{id}",
                        "linkTarget": "_self",
                        "endpointParameters": [
                            {
                                "name": "id",
                                "target": "path",
                                "value": {
                                    "dynamic": "id"
                                }
                            }
                        ]
                    }
                },
                {
                    "name": "testForm",
                    "type": "form",
                    "callback": "reloadData",
                    "componentAttributes": {
                        "endpoint": {
                            "service": "serviceName",
                            "namespace": "namespaceName",
                            "method": "methodName",
                            "resolve": false
                        },
                        "fields": [
                            {
                                "name": "fieldNameOne",
                                "component": "Input",
                                "componentAttributes": {}
                            },
                            {
                                "name": "fieldNameTwo",
                                "component": "Switch",
                                "componentAttributes": {}
                            }
                        ]
                    }
                }
            ],
            "fieldsGroup": [
                {
                    "name": "info",
                    "fields": [
                        {
                            "name": "someField",
                            "component": "Text",
                            "dependency": "dependencyOne",
                            "componentAttributes": {}
                        },
                        {
                            "name": "otherField",
                            "component": "Chip",
                            "componentAttributes": {
                                "borderColor": "grey"
                            }
                        }
                    ]
                }
            ],
            "conditions": {
                "matchWhen": [
                    [
                        {
                            "name": "isNotEmpty",
                            "field": [
                                "test",
                                "tes2"
                            ]
                        },
                        {
                            "name": "isNotEqualTo",
                            "field": "name",
                            "referenceValueType": "static",
                            "referenceValue": null
                        }
                    ]
                ]
            }
        },
        {
            "name": "someCardTwo",
            "component": "BaseCard",
            "fieldsGroup": [
                {
                    "name": "info",
                    "fields": [
                        {
                            "name": "someField",
                            "component": "Text",
                            "componentAttributes": {}
                        },
                        {
                            "name": "otherField",
                            "component": "Chip",
                            "componentAttributes": {
                                "borderColor": "grey"
                            }
                        }
                    ]
                }
            ],
            "conditions": {
                "matchWhen": [
                    [
                        {
                            "name": "isEqualTo",
                            "field": "name",
                            "referenceValue": "test"
                        }
                    ]
                ]
            }
        }
    ]
}